### ISSUE TITLE
cupy.all JitifyException: Runtime compilation failed

### ISSUE THREAD

**Issue by matt3o** [cupy/cupy#7560]
### Description

When I tried to use cp.all in my code, it just failed with an exception. Other cupy operations appear to work, I have been using e.g. the distance_transform_edt and it works perfectly fine.
Tried it with cupy 12 and 11.6 using first pip and then conda-forge for installation.
Below is a minimal version to trigger the error on my PC.

### To Reproduce

```py
import cupy as cp
test = cp.asarray([1,2,3])
cp.all(test > 0)
```

### Installation

Conda-Forge (`conda install ...`)

### Environment

```
OS                           : Linux-4.15.0-201-generic-x86_64-with-glibc2.17
Python Version               : 3.8.16
CuPy Version                 : 12.0.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.22.3
SciPy Version                : 1.9.1
Cython Build Version         : 0.29.34
Cython Runtime Version       : None
CUDA Root                    : /usr
nvcc PATH                    : /usr/bin/nvcc
CUDA Build Version           : 10020
CUDA Driver Version          : 11060
CUDA Runtime Version         : 10020
cuBLAS Version               : (available)
cuFFT Version                : 10102
cuRAND Version               : 10102
cuSOLVER Version             : (10, 3, 0)
cuSPARSE Version             : (available)
NVRTC Version                : (10, 2)
Thrust Version               : 100907
CUB Build Version            : <unknown>
Jitify Build Version         : 22da8f9
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce RTX 2080 Ti
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:08:00.0
Device 1 Name                : NVIDIA GeForce RTX 2080 Ti
Device 1 Compute Capability  : 75
Device 1 PCI Bus ID          : 0000:09:00.0
Device 2 Name                : NVIDIA GeForce RTX 2080 Ti
Device 2 Compute Capability  : 75
Device 2 PCI Bus ID          : 0000:42:00.0
Device 3 Name                : NVIDIA GeForce RTX 2080 Ti
Device 3 Compute Capability  : 75
Device 3 PCI Bus ID          : 0000:43:00.0

```

### Additional Information

```
(monai) mhadlich@i14s44:~/code/sliding-window-based-interactive-segmentation-of-volumetric-medical-images/user_guidance_study$ python
Python 3.8.16 (default, Mar  2 2023, 03:21:46) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cupy as cp
>>> test = cp.asarray([1,2,3])
>>> cp.all(test > 0)
---------------------------------------------------
--- JIT compile log for /tmp/tmp508p34md/58b059976d0fe66ef008f896a549a9a2aa9a71ec.cubin.cu ---
---------------------------------------------------
/usr/include/cuda.h(247): error: invalid redeclaration of type name "CUuuid_st"
__nv_nvrtc_builtin_header.h(1553): here

/usr/include/cuda.h(249): error: invalid redeclaration of type name "CUuuid"
__nv_nvrtc_builtin_header.h(1556): here

specializations/block_reduce_raking_commutative_only.cuh(98): warning: declaration does not declare anything

/tmp/tmp508p34md/58b059976d0fe66ef008f896a549a9a2aa9a71ec.cubin.cu(111): warning: variable "_J" was declared but never referenced

2 errors detected in the compilation of "/tmp/tmp508p34md/58b059976d0fe66ef008f896a549a9a2aa9a71ec.cubin.cu".

---------------------------------------------------
Traceback (most recent call last):
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 253, in _jitify_prep
    name, options, headers, include_names = jitify.jitify(source, options)
  File "cupy/cuda/jitify.pyx", line 64, in cupy.cuda.jitify.jitify
  File "cupy/cuda/jitify.pyx", line 90, in cupy.cuda.jitify.jitify
RuntimeError: Runtime compilation failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/_logic/truth.py", line 50, in all
    return a.all(axis=axis, out=out, keepdims=keepdims)
  File "cupy/_core/core.pyx", line 1174, in cupy._core.core._ndarray_base.all
  File "cupy/_core/core.pyx", line 1176, in cupy._core.core._ndarray_base.all
  File "cupy/_core/_routines_logic.pyx", line 8, in cupy._core._routines_logic._ndarray_all
  File "cupy/_core/_reduction.pyx", line 613, in cupy._core._reduction._SimpleReductionKernel.__call__
  File "cupy/_core/_reduction.pyx", line 369, in cupy._core._reduction._AbstractReductionKernel._call
  File "cupy/_core/_cub_reduction.pyx", line 704, in cupy._core._cub_reduction._try_to_call_cub_reduction
  File "cupy/_core/_cub_reduction.pyx", line 540, in cupy._core._cub_reduction._launch_cub
  File "cupy/_core/_cub_reduction.pyx", line 475, in cupy._core._cub_reduction._cub_two_pass_launch
  File "cupy/_util.pyx", line 67, in cupy._util.memoize.decorator.ret
  File "cupy/_core/_cub_reduction.pyx", line 246, in cupy._core._cub_reduction._SimpleCubReductionKernel_get_cached_function
  File "cupy/_core/_cub_reduction.pyx", line 231, in cupy._core._cub_reduction._create_cub_reduction_function
  File "cupy/_core/core.pyx", line 2257, in cupy._core.core.compile_with_cache
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 496, in _compile_module_with_cache
    return _compile_with_cache_cuda(
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 574, in _compile_with_cache_cuda
    ptx, mapping = compile_using_nvrtc(
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 322, in compile_using_nvrtc
    return _compile(source, options, cu_path,
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 287, in _compile
    options, headers, include_names = _jitify_prep(
  File "/home/mhadlich/anaconda3/envs/monai/lib/python3.8/site-packages/cupy/cuda/compiler.py", line 260, in _jitify_prep
    raise JitifyException(str(cex))
cupy.cuda.compiler.JitifyException: Runtime compilation failed
>>> test > 0
array([ True,  True,  True])
>>> (test > 0).dtype
dtype('bool')

```

**Comment by matt3o** [comment]
I got it to work on a different server, so I guess this has to be some weird interaction because of the way CUDA is installed (CUDA system-wide on 10.2 but locally on 11.6 for the cupy compilation). Any insights what do here what be cool, but apart from that I think we can close the issue and I'll try look into that discrepancy

**Comment by kmaehashi** [comment]
This issue happens when `libnvrtc.so.*` picks the wrong `libnvrtc-builtins.so.*`. I guess you are using 10.2 for the former but 11.6 for the latter. The fix depends on the setup, but removing cudatoolkit==11.6 from conda and using system-wide CUDA sounds good to me.

**Comment by leofang** [comment]
If you're using system-wide CUDA, I suggest to not install CuPy from conda-forge, but from PyPI (`pip install ...`). That conda-forge packages work with a system-wide CUDA is untested, unintended, and not guaranteed to work.

**Comment by matt3o** [comment]
Got it to run now. Maybe I explained it poorly, but I'll try to summarize it once more, so it can help someone else too.

Old server from an institute, running Cuda 10.2. I additionally installed CUDA 11.6 locally in my user directory and added it to $PATH. Now both CUDA versions where available from $PATH.
Additionally first I installed cupy via pip and got the installation to run on 11.6 (with 10.2 it would not compile and raise an error). However apparently the CUDA remainder in the PATH (or $LD_LIBRARY_PATH) led to the error above.
As a second step I removed the pip version and reinstalled via conda forge, from with the output above is. However I got the same erorr under pip as well.
I now got to work by removing the older CUDA version from PATH and LD_LIBRARY_PATH.
